src/main/java/org/chainoptim/desktop/core/main/service/NavigationService.java

```diff
@@ -46,6 +57,25 @@ public void switchView(String viewKey) {
         // Get view from cache or load it
         Node view = viewCache.computeIfAbsent(viewKey, this::loadView);
 
+        //Apply CSS
+        String cssPath = cssMap.get(viewKey);
+        if (cssPath != null) {
+            URL cssURL = getClass().getResource(cssPath);
+            if (cssURL != null) {
+                String css = cssURL.toExternalForm();
+                if (view instanceof Parent) {
```

Comment:
Separation of concerns: take this into a separate function. Also, get rid of nesting by checking cssPath == null etc